src/Adapter/AbstractAdapter.php

```diff
@@ -4,5 +4,21 @@
 
 class AbstractAdapter implements AdapterInterface
 {
+    protected $trainingData;
+
+    public function getTrainingData()
+    {
+        return $this->trainingData;
+    }
+
+    public function train()
+    {
+        // TODO: Implement train() method.
+    }
+
+    public function predicte()
```

Comment:
and for an unified interface we could have a call() our exec() method that would call the appropriate method.
But we may need to abstract adapters, one for Prediction and one for Classification...